SystemWrapper/DateTimeWrap.cs

```diff
@@ -526,16 +536,23 @@ public int DaysInMonth(int year, int month)
             return DateTime.DaysInMonth(year, month);
         }
 
-        /// <inheritdoc />
+        /// <summary>
+        /// Returns a value indicating whether this instance is equal to the specified IDateTimeWrap instance.
+        /// </summary>
+        /// <param name="value">A IDateTimeWrap instance to compare to this instance. </param>
+        /// <returns> <c>true</c> if the value parameter equals the value of this instance; otherwise, false.</returns>
         public bool Equals(IDateTime value)
         {
-            return DateTimeInstance.Equals(value);
```

Comment:
Unless I'm missing something, were we comparing a `DateTime` with a `DateTimeWrap`? If so, that would have always returned `false` I think? I can confirm that at the moment a DateTime instance isn't equal to itself. The change should fix it - but I'm keen to confirm I haven't missed something subtle here.